### DISCUSSION TITLE
Unable to push my code

### DISCUSSION THREAD

**Discussion by rishavchatterjeeTWC** [community/community#128999] [Repositories]
### Select Topic Area

General

### Body

I am unable to push my code to the repo on github. I have admin access to the repo. But while pushing the code I get- 
_remote: Write access to repository not granted._

**Comment by buu-huu** [comment]
This most likely indicates a permission issue. Do you authenticate via HTTPS with a GitHub personal access token?

**Comment by rishavchatterjeeTWC** [comment]
Hi, I have checked the access I have for this repo. It is admin access. And yes, I have authenticated using HTTPS with a GitHub personal access token.

**Reply by buu-huu** [reply]
Is your access token still valid or did it expire? Try to generate a new one with the Repo permissions scope.

If that doesn't work, try the following:

1. Go to the repositories settings
2. Then go to Actions -> General
3. Scroll down to "Workflow Permissions"
4. Set to "Read and write permissions"

**Comment by rishavchatterjeeTWC** [comment]
Hi, I already have 'Read and write permissions' for Workflow Permissions. Are there any other permissions that need to be updated?

**Comment by hamburnyog** [comment]
Are you using fine-grained tokens? you could update the token then check if these are set:

![image](https://github.com/user-attachments/assets/1b994003-6882-40d8-a3e6-dc270b6cd828)

I think you only need content permission but you can also add administration.
